crates/iceberg/src/transform/bucket.rs

```diff
@@ -167,6 +167,16 @@ impl TransformFunction for Bucket {
                 .downcast_ref::<arrow_array::TimestampMicrosecondArray>()
                 .unwrap()
                 .unary(|v| self.bucket_timestamp(v)),
+            DataType::Time64(TimeUnit::Nanosecond) => input
+                .as_any()
+                .downcast_ref::<arrow_array::Time64NanosecondArray>()
+                .unwrap()
+                .unary(|v| self.bucket_time(v / 1000)),
```

Comment:
there was an issue with rounding when performing division for the hour transform (https://github.com/apache/iceberg-rust/pull/1146) 
but since we're dividing by `1000`, i dont think the same issue applies here